src/cascadia/TerminalSettingsModel/TerminalSettings.cpp

```diff
@@ -306,6 +306,8 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
             _TabColor = static_cast<winrt::Microsoft::Terminal::Core::Color>(colorRef);
         }
 
+        _EnvironmentVariables = profile.EnvironmentVariables();

```

Comment:
Ohhhhh

this one we might actually want to *copy*, I think? The version stored in the profile is mutable (!)

Either that, or we should use a MapView on profile's public API and a Map internally :)